torch/distributed/fsdp/_common_utils.py

```diff
@@ -283,7 +283,7 @@ def module_fn(module, prefix, tree_level, param_to_fqns):
         ):
             local_fqns = (
                 param._fqns
-                if type(param) is flat_param_file.FlatParameter
+                if isinstance(param, flat_param_file.FlatParameter)
```

Comment:
For context, I think we had to change these to `type(param) is FlatParameter` because otherwise other tensor subclasses with `_is_param = True` return `True` from the `instance(tensor, FlatParameter)` check. See this PR description for more details: https://github.com/pytorch/pytorch/pull/85038

Possibly your new override of `__instancecheck__()` resolves that issue?

Reply:
Yes it does! This is also an interesting failure mode for the `_is_param` trick from Parameter which might explain why @jbschlosser found it was BC-breaking to do the `_is_param` trick universally.